packages/next/server/node-polyfill-fetch.js

```diff
@@ -14,7 +25,27 @@ if (!global.fetch) {
     } else if (!opts.agent) {
       opts.agent = agent
     }
-    return fetch(url, opts, ...rest)
+    let canBeCached =
+      process.env.NODE_ENV === 'production' &&
+      (!opts ||
+        (opts.method === 'GET' &&
+          !opts.body &&
+          !opts.cache &&
+          !opts.signal &&
+          (!opts.headers || !opts.headers.pragma)))
```

Comment:
In `RequestInit` (fetch option), headers can either be a plain object or a `Header` object:

```js
const header1 = { pragma: 'no-cache' }
header1.pragma
// 'no-cache'

const header2 = new Headers({ pragma: 'no-cache' })
header2.pragma
// undefined
```

So it is not safe.